public/app/features/alerting/components/OptionElement.tsx

```diff
@@ -0,0 +1,53 @@
+import React, { FC } from 'react';
+import { FormAPI, Input, InputControl, Select, Switch, TextArea } from '@grafana/ui';
+import { Option } from '../../../types';
+
+interface Props extends Pick<FormAPI<any>, 'register' | 'control'> {
+  option: Option;
+}
+
+export const OptionElement: FC<Props> = ({ control, option, register }) => {
+  const modelValue = `settings.${option.propertyName}`;
+  switch (option.element) {
```

Comment:
nit: would it be a good idea to make a `console.error` or display an error toast if and element type not supported in the frontend is passed here? Basically adding a default case that does that. My thinking behind it is if we in the future start sending an element from the backend that we don't support on frontend it will be easier to detect then if it is just not rendered.